hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -330,12 +358,24 @@ public void clearPartitionsOf(Predicate<RecordStore> predicate, int partitionId)
         while (partitionIterator.hasNext()) {
             RecordStore partition = partitionIterator.next();
             if (predicate.test(partition)) {
-                partition.clearPartition(false, false);
+                partition.clearPartition(onShutdown, onRecordStoreDestroy);
+                if (onShutdown || onReset) {
+                    destroyEventJournalOf(partition);
```

Comment:
I think in the case of `onShutdown=true`, at this point the event journal has already been destroyed by  `partition.clearPartition()`. This is not an issue though as long as the destroy method is idempotent.

Also, maybe we should call destroy on the `RecordStoreMutationObserver` rather than calling it directly on the event journal. This needs the `createRecordStoreMutationObservers` to be changed to `getOrCreate...`.

Reply:
Good point. 
Removed lines 362 -363 above and relied on `clearPartition`.